app/services/foreman/plugin.rb

```diff
@@ -351,6 +351,11 @@ def parameter_filters(klass)
       @parameter_filters.fetch(klass.is_a?(Class) ? klass.name : klass, [])
     end
 
+    # Add concerns to a host
+    def extend_host(concern_module)
```

Comment:
I saw the request - I just think it's best to leave this kind of things to Rails. I don't necessarily see it as a bad thing that plugin devs know about objects in Foreman, in fact they pretty much have to know about them. Anyway, not that out of bounds that I would block this PR for that, I'll probably just rely on Rails for extending core models & that's it